drivers/npmx/shell/common.c

```diff
@@ -228,3 +228,35 @@ bool check_pin_configuration_correctness(const struct shell *shell, int8_t gpio_
 
 	return true;
 }
+
+bool charger_disabled_check(const struct shell *shell, npmx_charger_t *charger_instance,
+			    const char *help)
+{
+	uint32_t modules_mask;
+	npmx_error_t err_code = npmx_charger_module_get(charger_instance, &modules_mask);
+
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "charger module status");
+		return false;
+	}
+
+	if ((modules_mask & NPMX_CHARGER_MODULE_CHARGER_MASK) != 0) {
+		shell_error(shell, "Error: charger must be disabled to set %s.", help);
+		return false;
+	}
+	return true;
+}
+
+npmx_adc_t *adc_instance_get(const struct shell *shell)
```

Comment:
I think functions getting instance of a peripheral should be in file for such peripheral, not in common. I know, that charger needs adc instance in some commands, and adc needs charger instance, but maybe including other's peripherals' header file would solve the problem?